lib/mrsk/commands/app.rb

```diff
@@ -82,20 +82,22 @@ def execute_in_new_container_over_ssh(*command, host:)
   end
 
 
-  def current_container_id
-    docker :ps, "--quiet", *filter_args
+  def current_running_container_id
+    docker :ps, "--quiet", *filter_args(status: :running), "--latest"
   end
 
   def container_id_for_version(version)
     container_id_for(container_name: container_name(version))
   end
 
   def current_running_version
-    # FIXME: Find more graceful way to extract the version from "app-version" than using sed and tail!
+    list_versions("--latest", status: :running)
+  end
+
+  def list_versions(*docker_args, status: nil)
     pipe \
-      docker(:ps, *filter_args, "--format", '"{{.Names}}"'),
-      %(sed 's/-/\\n/g'),
-      "tail -n 1"
+      docker(:ps, *filter_args(status: status), *docker_args, "--format", '"{{.Names}}"'),
+      %(grep -oP "(?<=\-)[^-]+$") # Extract SHA from "service-role-dest-SHA"
```

Comment:
`-P` is a GNU grep arg. Got a more portable alternative?

Reply:
TIL. The reason for using `-P` was to rely on lookbehind to capture only the SHA. I come up with a combination of grep + cut to achieve it instead.